packages/swapper/src/swappers/ButterSwap/swapperApi/checkTradeStatus.ts

```diff
@@ -0,0 +1,50 @@
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+
+import type { CheckTradeStatusInput, TradeStatus } from '../../../types'
+import { getBridgeInfoBySourceHash } from '../xhr'
+
+// See: https://docs.butternetwork.io/butter-swap-integration/butter-api-for-swap-data/get-swap-history-by-source-hash
+// Example endpoint: https://bs-router-v3.chainservice.io/api/queryBridgeInfoBySourceHash?sourceHash=0x...
+
+// State mapping: 0 = pending, 1 = complete, 6 = refunded/failed
+const BUTTER_SWAP_STATES = (() => ({
+  Pending: 0,
+  Confirmed: 1,
+  Failed: 6,
+}))()
+
+export const checkTradeStatus = async (input: CheckTradeStatusInput): Promise<TradeStatus> => {
+  const { txHash } = input
```

Comment:
suggestion: we should be able to skip all this logic as an early return and use `checkSolanaSwapStatus` and `checkEvmSwapStatus` for same-chain Txs

Reply:
ser `for same-chain Txs` means the destination chain would be the source chain